### ISSUE TITLE
[bug][error] Chart with lines and bars produces repair file dialog in Powerpoint

### ISSUE THREAD

**Issue by kornarakis** [gitbrent/PptxGenJS#1013]
Type: Bug
Library Version: 3.8.0
Powerpoint Version: 365
Framework used: VueJS

### Observed Behavior

The following chart was working perfectly but now shows repair dialog in Powerpoint and some times Powerpoint is unable to repair the file.

### Steps to Reproduce

Demo code

```js
const pptx = new Pptxgen();

pptx.author = "The author name";
pptx.company = "The company name";
pptx.revision = "1";
pptx.subject = "A subject here";
pptx.title = "A title here";
pptx.defineLayout({
  name: "Custom",
  width: 11.694,
  height: 8.264,
});
pptx.layout = "Custom";

let slide1= pptx.addSlide();
slide1.addChart(
  [
    {
      type: "bar",
      data: [
        {
          name: "one",
          labels: [
            "label 1",
            "label 2",
            "label 3",
            "label 4",
            "label 5",
            "label 6",
            "label 7",
            "label 8",
            "label 9",
            "label 10",
            "label 11",
          ],
          values: [54, 69, 223, 401, 501, 731, 901, 639, 424, 685, 447],
        },
      ],
      options: {
        chartColors: ["FED9FF"],
      },
    },
    {
      type: "line",
      data: [
        {
          name: "two",
          labels: [
            "label 1",
            "label 2",
            "label 3",
            "label 4",
            "label 5",
            "label 6",
            "label 7",
            "label 8",
            "label 9",
            "label 10",
            "label 11",
          ],
          values: [13, 28, 67, 94, 107, 127, 143, 100, 66, 59, 22],
        },
      ],
      options: {
        lineSize: 0,
        lineDataSymbol: "circle",
        lineDataSymbolSize: 10,
        lineDataSymbolLineSize: 2,
        lineDataSymbolLineColor: "0096FF",
        chartColors: ["FFFFFF"],
        secondaryValAxis: true,
        secondaryCatAxis: true,
      },
    },
    {
      type: "line",
      data: [
        {
          name: "three",
          labels: [
            "label 1",
            "label 2",
            "label 3",
            "label 4",
            "label 5",
            "label 6",
            "label 7",
            "label 8",
            "label 9",
            "label 10",
            "label 11",
          ],
          values: [80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80],
        },
      ],
      options: {
        lineSize: 2,
        lineDataSymbol: "none",
        chartColors: ["D6D6D6"],
        secondaryValAxis: true,
        secondaryCatAxis: true,
      },
    },
    {
      type: "line",
      data: [
        {
          name: "four",
          labels: [
            "label 1",
            "label 2",
            "label 3",
            "label 4",
            "label 5",
            "label 6",
            "label 7",
            "label 8",
            "label 9",
            "label 10",
            "label 11",
          ],
          values: [120, 120, 120, 120, 120, 120, 120, 120, 120, 120, 120],
        },
      ],
      options: {
        lineSize: 2,
        lineDataSymbol: "none",
        chartColors: ["53585F"],
        secondaryValAxis: true,
        secondaryCatAxis: true,
      },
    },
  ],

  {
    x: 1.15,
    y: 2.24,
    w: 9.39,
    h: 4.49,
    legendPos: "b",
    barGapWidthPct: 50,
    showLegend: true,
    showTitle: false,
    catAxisLabelFontSize: 10,
    valAxes: [
      {
        showValAxisTitle: true,
        valAxisTitle: "something",
      },
      {
        showValAxisTitle: true,
        valAxisTitle: "something here",
        valGridLine: {
          style: "none",
        },
      },
    ],
    catAxes: [
      {},
      {
        catAxisHidden: true,
      },
    ],
  }
);

pptx.writeFile({ fileName: "demo.pptx" }).then(() => {});
```

**Comment by gitbrent** [comment]
[Ed]: Related to Issue #985?

**Comment by ReimaFrgos** [comment]
This issue appeared in 3.7. Can see it working fine in 3.6. I'm not sure atm why, but there is a simple fix to get you going.

The problem appears to be that barGrouping is not being set to a default value with the multi chart which is causing it to trigger the repair for the missing data. The repair sets this back to the default and the slide looks fine.

Temp Fix: add `barGrouping: "clustered"`

in context:
```
    x: 1.15,
    y: 2.24,
    w: 9.39,
    h: 4.49,
    barGrouping: "clustered"
    legendPos: "b",
    barGapWidthPct: 50,
    showLegend: true,
    showTitle: false,
    catAxisLabelFontSize: 10,
```

**Comment by hhq365** [comment]
thanks

**Comment by hhq365** [comment]
> This issue appeared in 3.7. Can see it working fine in 3.6. I'm not sure atm why, but there is a simple fix to get you going.
> 
> The problem appears to be that barGrouping is not being set to a default value with the multi chart which is causing it to trigger the repair for the missing data. The repair sets this back to the default and the slide looks fine.
> 
> Temp Fix: add `barGrouping: "clustered"`
> 
> in context:
> 
> ```
>     x: 1.15,
>     y: 2.24,
>     w: 9.39,
>     h: 4.49,
>     barGrouping: "clustered"
>     legendPos: "b",
>     barGapWidthPct: 50,
>     showLegend: true,
>     showTitle: false,
>     catAxisLabelFontSize: 10,
> ```

thank you!
I have been waiting 4 months for this answer.you fixed it.Now I can update to the latest version.

**Comment by gitbrent** [comment]
Thanks for reporting this @kornarakis 

The issue is fixed in upcoming release 3.11.0
